src/main/java/com/palantir/gradle/gitversion/GitImpl.java

```diff
@@ -25,33 +26,55 @@
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Mimics git describe by using rev-list to support versions of git < 1.8.4.
  */
-class NativeGitDescribe implements GitDescribe {
-    private static final Logger log = LoggerFactory.getLogger(NativeGitDescribe.class);
+class GitImpl implements Git {
+    private static final Logger log = LoggerFactory.getLogger(GitImpl.class);
 
     private static final Splitter LINE_SPLITTER =
             Splitter.on(System.getProperty("line.separator")).omitEmptyStrings();
     private static final Splitter WORD_SPLITTER = Splitter.on(" ").omitEmptyStrings();
 
     private final File directory;
 
-    NativeGitDescribe(File directory) {
+    GitImpl(File directory) {
+        if (!gitCommandExists()) {
+            throw new RuntimeException("Git not found in project");
+        }
+        this.directory = directory;
```

Comment:
I would replace this whole block with `this(directory, false)` rather than repeating code.